### PR TITLE
[doc] Update D2 checklist 

### PR DESCRIPTION
**Pull request by msfschaffner**
This addresses #13148.

Note that several checklists were a bit out of date and had to be aligned beyond the D2 updates in the first commit.

Note: some important security IP checklist updates have been placed into separate commits to ease the review.

### CODE REVIEW

File: doc/project/checklist.md [hunk 1]
```diff
@@ -117,28 +122,29 @@ Any lint waiver files have been reviewed.
...
-The implementation of security countermeasures can be delayed until D3 if:
-- The addition of those countermeasures would not negatively impact PPA by more than 10%, and
-- The software interface would not be materially affected by those countermeasures
+An area check has been completed either with an FPGA or ASIC synthesis flow.
 
-Where the area impact of countermeasures can be reliably estimated, it is acceptable to insert dummy logic at D2 in order to meet the above criteria.
+### TIMING_CHECK
 
-### SEC_RND_CNST
+A timing check has been completed either with an FPGA or ASIC synthesis flow.
```
Comment (sriyerg) [inline doc/project/checklist.md:140]: We only run and publish these numbers from the synthesis flow, right?

**Review by vogelpi — APPROVED** [review]
This looks good, thanks @msfschaffner !

File: doc/project/checklist.md (hunk 1 above)
Comment (msfschaffner) [reply doc/project/checklist.md:140]: Yeah the dashboard only publishes the numbers from the ASIC flow at the moment.
The checklist is supposed to be generic however, hence we've mentioned FPGA or ASIC synthesis flow without referring to a specific tool.

**Comment by msfschaffner** [comment]
Thanks for the reviews - merging this in so it does not get stale.
If you spot something that needs to be corrected, please let me know (or send me a PR ;)).

**Pull request merged**